substrate/frame/assets/src/functions.rs

```diff
@@ -1029,8 +1030,21 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			.collect::<Vec<_>>()
 	}
 
+	/// Ensure the asset is live or live and with no privileges.
+	pub(super) fn ensure_live_asset(
+		d: &AssetDetails<T::Balance, T::AccountId, DepositBalanceOf<T, I>>,
+	) -> DispatchResult {
+		ensure!(
+			d.status == AssetStatus::Live || d.status == AssetStatus::LiveAndNoPrivileges,
+			Error::<T, I>::AssetNotLive
+		);
+		Ok(())
+	}
+
 	/// Reset the team for the asset with the given `id`.
 	///
+	/// If the asset status is `LiveAndNoPrivileges` then it is changed to `Live`.
```

Comment:
I only improved the doc: https://github.com/paritytech/polkadot-sdk/pull/4150/commits/d0a8ab20f6d1357787f4ad253256185ef77d4d7f

I think it is fine to keep the trait as it is, because `reset_team` is quite authoritative by itself, and caller should know why it can or cannot reset the team.